azure-pipelines.yml

```diff
@@ -70,48 +70,40 @@ jobs:
       JAVA_TOOL_OPTIONS: " -Xmx8g"
       DOCKER_BUILDKIT: 1
     container:
-      image: uchimera.azurecr.io/cccs/dev/projector-intellij-ce:trino_20240716153915_b10530
+      image: uchimera.azurecr.io/cccs/dev/trino-build:452_feature_new-dev-environment
       options: --entrypoint="" --user=0
       endpoint: uchimera
       volumes:
         - "${{ variables.m2RepoNodePath }}:${{ variables.m2RepoContainerPath }}"
     steps:
       - task: DownloadSecureFile@1
-        name: cccs_stratus_m2_settings
-        displayName: "Download settings.xml"
+        name: mavenSettings
+        displayName: Download settings.xml
         inputs:
-          secureFile: "settings.xml"
-      - task: Bash@3
-        name: move_settings_to_m2
-        displayName: "Move settings to .m2"
-        inputs:
-          targetType: "inline"
-          script: |
-            mkdir -p "${HOME}/.m2"
-            cp "$(cccs_stratus_m2_settings.secureFilePath)" "${HOME}/.m2/settings.xml"
+          secureFile: settings.xml
       - task: Bash@3
         name: setBranchName
+        displayName: Parse Source Control Branch Name
         inputs:
           targetType: 'inline'
           script: |
             BRANCH_NAME=$(echo "$SYSTEM_PULLREQUEST_SOURCEBRANCH $BUILD_SOURCEBRANCH" | sed -r 's/^\s*(refs\/heads\/)?(\S*).*$/\2/' | sed 's/\//_/g')
             echo "##vso[task.setvariable variable=BRANCH_NAME]$BRANCH_NAME"
-        displayName: Parse Source Control Branch Name
       - task: Bash@3
-        name: BuildTrino
+        name: maven
         displayName: Maven build
         inputs:
           targetType: 'inline'
           script: |
-            $(Build.SourcesDirectory)/mvnw clean install -e -DskipTests -pl '!:trino-docs' -Dmaven.repo.local=${{ variables.m2RepoContainerPath }}
+            $(Build.SourcesDirectory)/mvnw clean install -e -DskipTests -pl '!:trino-docs' --settings "$(mavenSettings.secureFilePath)" -Dmaven.repo.local=${{ variables.m2RepoContainerPath }}
```

Comment:
I'm having a hard time figuring out what this $(mavenSettings.secureFilePath) will be? Should this file not be in "${HOME}/.m2/settings.xml" if say we want to add another repo? For example, right now we're referring to https://github.com/CybercentreCanada/trino/blob/9aa9fe945f64d14e54bb143a6c5658002609934d/pom.xml#L2301-L2302 but I don't think we authenticate to to it or use it at all. I'm also wondering if other commands that depend on the file will know to find it

Reply:
That's the output of the `DownloadSecureFile@1` task that starts at line 79. `mavenSettings` is the task's name. So that task injects the secure file somewhere in the container and we just use it. If you look at lines 84-91 of the old version, you'll see we used to have a `Bash@3` task that copied that file to `${HOME}/.m2/settings.xml`. With this change, that task is no longer required.